packages/web-components/fast-foundation/src/date-picker/date-picker.ts

```diff
@@ -0,0 +1,1262 @@
+     * @param force - Should the toggle ignore the overFlyout boolean
+     * @public
+     */
+    public toggleFlyout(force: boolean = false): void {
+        this[`${this.flyoutOpen ? "close" : "open"}Flyout`](force);
+    }
+
+    public handleFlyoutKeydown(e: KeyboardEvent): void {
+        if (e && e.key && e.key === keyEscape) {
+            e.preventDefault();
+            this.closeFlyout();
+            this.textField.control.focus();
+        }
+    }
+
+    /**
+     * Should show the calendar
+     * @public
+     */
+    @observable
+    public showCalendar: boolean = true;
+
+    /**
+     * Should dshow the month picker when date selecting
+     * @public
+     */
+    @observable
+    public showMonthPicker: boolean = false;
+
+    /**
+     * Method for displaying the month picker
+     * @param open - Should open the month picker
+     * @returns void
+     * @public
+     */
+    public monthPickerDisplay(open: boolean = true): void {
+        if (this.type === "month") {
+            return;
+        }
+
+        this.showMonthPicker = open;
+        if (this.type === "datetime-local") {
+            this.showCalendar = !open;
+        }
+    }
+
+    /**
+     * Should show the year picker when date selecting
+     * @public
+     */
+    @observable
+    public showYearPicker: boolean = false;
+
+    /**
+     * Method for displaying the year picker
+     * @param open - should show the year picker
+     * @returns void
+     */
+    public yearPickerDisplay(open: boolean = true, e?: KeyboardEvent): boolean {
+        if (!e || e.key === keyEnter) {
+            if (this.type === "month" || this.type === "year") {
+                return true;
+            }
+
+            this.showYearPicker = open;
+            if (this.type === "date" || this.type === "datetime-local") {
+                this.showMonthPicker = !open;
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Handler for selected date in the calendar
+     * @param date - The date clicked
+     * @public
+     */
+    public handleDateClicked(event: MouseEvent): void {
+        const { day, month, year } = event.detail as any;
+        this.setValue({ day, month, year });
+    }
+
+    /**
+     * Handler for selecting a month in the month picker
+     * @param month - The month clicked
+     * @public
+     */
+    public handleMonthClicked(month: number, year: number): void {
+        this.calendarMonth = month;
+        this.calendarYear = year;
+
+        if (this.type === "month") {
+            this.setValue({ month, year });
+        }
+
+        if (this.type === "datetime-local") {
+            this.monthPickerDisplay(false);
+        }
+    }
+
+    /**
+     * Handler for keyboard actions for a month
+     * @param event - keyboard event
+     * @returns true
+     */
+    public handleMonthKeyup(month: number, year: number, event: KeyboardEvent): void {
```

Comment:
Not sure if this prevents default on all keystrokes or not with the ".bind" syntax.

Reply:
Updated this.